File: 0541-reverse-string-ii/0541-reverse-string-ii.cpp
class Solution {
public:

    void reverse(string& s, int start, int end) {
        while (start < end) {
            char temp = s[start];
            s[start] = s[end];
            s[end] = temp;
            start++;
            end--;
        }
    }

    string reverseStr(string s, int k) {
        int n = s.length();
        
        for (int i = 0; i < n; i += 2 * k) {
            int start = i;
            int end = min(i + k - 1, n - 1); 
            
            reverse(s, start, end);
        }
        
        return s;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0541-reverse-string-ii/0541-reverse-string-ii.java
class Solution {
    public String reverseStr(String arr, int k) {
        int n = arr.length();
        char[] s = arr.toCharArray();

        for(int i=0; i<n ; i+= 2*k) {
            int start = i;
            int end = Math.min(i+k-1 , n-1);

            while(start<end) {
                char temp = s[start];
                s[start] = s[end];
                s[end] = temp;
                start++ ;
                end-- ;
            }
        }
        return new String(s);
    }
}